viewer/src/main/java/org/kaazing/monitoring/agrona/viewer/task/MetricsTaskImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.kaazing.monitoring.agrona.viewer.task;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.kaazing.monitoring.agrona.viewer.MetricsViewer;
+import org.kaazing.monitoring.reader.api.MessagesCollector;
+import org.kaazing.monitoring.reader.api.MetricsCollector;
+
+/**
+ * Implementation for the GetMetricsTask abstraction
+ *
+ */
+public class MetricsTaskImpl implements MetricsTask{
+    private MetricsCollector metricsCollector;
+    private MessagesCollector messagesCollector;
+    private String fileName;
+    private ScheduledFuture<?> task;
+
+    public MetricsTaskImpl(String fileName, ScheduledExecutorService taskExecutor, MetricsCollector metricsCollector, MessagesCollector messagesCollector) {
+        this.fileName = fileName;
+        this.metricsCollector = metricsCollector;
+        this.messagesCollector = messagesCollector;
+
+        @SuppressWarnings("unused")
+        ScheduledFuture<?> task = taskExecutor.scheduleAtFixedRate(() -> {
```

Comment:
Thank you. Your observations are correct, the class attribute should have been updated here. Will submit a new commit with the changes.